### ISSUE TITLE
Not working correctly

### ISSUE THREAD

**Issue by JZZQuant** [EvanOman/AuctionAlgorithm#2]
I tried using this to solve a simple  assignment problem.
http://www.universalteacherpublications.com/univ/ebooks/or/ch6/hungar.htm
it doesn't seem to be converging to the correct answer
<img width="639" alt="screen shot 2017-12-03 at 12 56 19 pm" src="https://user-images.githubusercontent.com/4330556/33523323-b0b871b2-d829-11e7-8eeb-582895a513eb.png">

**Comment by EvanOman** [comment]
Thanks for finding this bug!

Unfortunately it looks like there are several real issues here (see issue #1) that were introduced in my last pre-push refactoring.

If I get some time I will definitely get to the bottom of this (and add a few correctness tests for good measure)

**Comment by JZZQuant** [comment]
this could fix:
change line number 59 to :  double epsilon =  1.0/(N+1);
get rid of the outer while loop at line 62, you dont need it.

**Comment by WayneTian-42** [comment]
> this could fix: change line number 59 to : double epsilon = 1.0/(N+1); get rid of the outer while loop at line 62, you dont need it.

I agree with you. The result of this modification is better than the original code.

**Comment by fdiedler** [comment]
@JZZQuant @WayneTian-42 
Hi,

Can you help me to modify the code for min assignment problem ? 
https://github.com/EvanOman/AuctionAlgorithmCPP/issues/3

Thanks,

**Comment by EvanOman** [comment]
You were right — the epsilon-scaling loop exited one phase too early, so tie-heavy matrices could come back suboptimal. Fixed in the recent rewrite; your 4x4 now solves to 78, the true optimum. Gory details in docs/BUGS.md. Thanks for the report (a decade later)!
